Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py

```diff
@@ -3170,6 +3222,128 @@ def run_remote_script_command(client: Client, args: dict) -> CommandResults:
+    results = client.get_remote_script_results_request(computer_names, task_ids)
+    file_results = []
+    for result in results:
+        if result.get("downloadUrl", ""):
+            response = requests.get(url=result.get("downloadUrl"))
+            zip_file_data = response.content
+            file_results.append(fileResult(filename=result.get('fileName', ''),
+                                data=zip_file_data, file_type=EntryType.ENTRY_INFO_FILE))
+            context_entries.append({
+                'taskId': result.get("taskId"),
+                'fileName': result.get("fileName"),
+                'downloadUrl': result.get("downloadUrl")
+            })
+    return [CommandResults(
+        readable_output=tableToMarkdown("SentinelOne - Get Remote Scripts Results", results, headers=headers, removeNull=True),
+        outputs_prefix="SentinelOne.RemoteScriptResults",
+        outputs_key_field='taskId',
+        outputs=context_entries,
+        raw_response=results),
+        *file_results
+    ]
+
+
+def run_polling_command(client: Client, cmd: str, args: Dict[str, Any]):
+    """
+    This command is combination of the **Run Remote Script**, **Remote Script Status Check**,
+    and **Remote Script Results** commands.
+    The polling command continuously polls the data until the status of the executed remote script is marked as complete,
+    and then it returns the results of that remote script.
+    Args:
+        cmd (str): The command name.
+        client (Client): SentinelOne API client.
+        args (Dict[str, Any]): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+    """
+    ScheduledCommand.raise_error_if_not_supported()
+    interval = arg_to_number(args.get('interval', 60))
+    timeout = arg_to_number(args.get('timeout', 600))
+    if 'parentTaskId' not in args:
+        command_results = run_remote_script_command(client, args)
+        output = command_results.raw_response
+        if isinstance(output, dict):
+            parent_task_id = output.get("parentTaskId")
+            args['parentTaskId'] = parent_task_id
+        scheduled_command = schedule_command(interval, timeout, cmd, args)
+        command_results.scheduled_command = scheduled_command
+        return command_results
+
+    parent_task_id = args.get('parentTaskId')
+    status_args = {"parentTaskId": parent_task_id}
+    status_check_command_results = get_remote_script_status(client, status_args)
+    status_outputs = status_check_command_results.raw_response
+    script_completed = False
+    task_ids = []
+    if status_outputs and isinstance(status_outputs, list):
+        for output in status_outputs:
+            if isinstance(output, dict):
+                if output.get("status") != "completed":
+                    script_completed = False
+                    break
+                task_ids.append(output.get("id"))
+                script_completed = True
```

Comment:
As we saw that there can be multiple status outputs, it might be that not all of them are 'True', lets handle it properly, and evaluate all statuses' completion combined.

Reply:
done